externals/kyuubi-flink-sql-engine/src/main/java/org/apache/kyuubi/engine/flink/config/entries/ExecutionEntry.java

```diff
@@ -0,0 +1,382 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine.flink.config.entries;
+
+import static org.apache.kyuubi.engine.flink.config.EngineEnvironment.EXECUTION_ENTRY;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.flink.api.common.restartstrategy.RestartStrategies;
+import org.apache.flink.api.common.time.Time;
+import org.apache.flink.streaming.api.TimeCharacteristic;
+import org.apache.flink.table.api.EnvironmentSettings;
+import org.apache.flink.table.descriptors.DescriptorProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Configuration of a table program execution. This class parses the `execution` part in an
+ * environment file. The execution describes properties that would usually be defined in the
+ * ExecutionEnvironment/StreamExecutionEnvironment/TableEnvironment or as code in a Flink job.
+ *
+ * <p>All properties of this entry are optional and evaluated lazily. TODO remove the useless
+ * properties
+ */
+public class ExecutionEntry extends ConfigEntry {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ExecutionEntry.class);
+
+  public static final ExecutionEntry DEFAULT_INSTANCE =
+      new ExecutionEntry(new DescriptorProperties(true));
+
+  public static final String EXECUTION_PLANNER = "planner";
+
+  public static final String EXECUTION_PLANNER_VALUE_OLD = "old";
```

Comment:
The legacy planer is not recommended for a long time, and has been remove in recent version, can we just drop support?